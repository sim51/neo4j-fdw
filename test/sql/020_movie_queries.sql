\a
SELECT * FROM movie ORDER BY id ASC LIMIT 10;
SELECT * FROM movie m WHERE m.released > 2000 ORDER BY m.released DESC, m.id DESC LIMIT 10;
SELECT * FROM movie m WHERE m.title = 'The Matrix' ORDER BY m.id ASC;
SELECT * FROM movie m WHERE m.title LIKE '%Matrix%' ORDER BY m.id ASC;
SELECT * FROM movie m WHERE m.title ILIKE '%matrix%' ORDER BY m.id ASC;
SELECT count(*) FROM movie m;
SELECT count(*) FROM movie m WHERE m.released > 2000;
SELECT avg(released) FROM movie m WHERE m.title like '%Matrix%';
SELECT title, released, tagline FROM movie m WHERE m.title IN ('The Matrix', 'Top Gun') AND m.released > 1980 ORDER BY m.id ASC;
SELECT title, released, tagline FROM movie m WHERE NOT m.title IN ('The Matrix', 'Top Gun') AND m.released > 1980 ORDER BY m.id ASC;
