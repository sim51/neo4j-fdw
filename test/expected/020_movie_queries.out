\a
SELECT * FROM movie ORDER BY id ASC LIMIT 10;
id|title|released|tagline
0|The Matrix|1999|Welcome to the Real World
9|The Matrix Reloaded|2003|Free your mind
10|The Matrix Revolutions|2003|Everything that has a beginning has an end
11|The Devil's Advocate|1997|Evil has its winning ways
15|A Few Good Men|1992|In the heart of the nation's capital, in a courthouse of the U.S. government, one man will stop at nothing to keep his honor, and one will stop at nothing to find the truth.
29|Top Gun|1986|I feel the need, the need for speed.
37|Jerry Maguire|2000|The rest of his life begins now.
46|Stand By Me|1986|For some, it's the last real taste of innocence, and the first real taste of life. But for everyone, it's the time that memories are made of.
52|As Good as It Gets|1997|A comedy from the heart that goes for the throat.
56|What Dreams May Come|1998|After life there is more. The end is just the beginning.
(10 rows)
SELECT * FROM movie m WHERE m.released > 2000 ORDER BY m.released DESC, m.id DESC LIMIT 10;
id|title|released|tagline
105|Cloud Atlas|2012|Everything is connected
128|Ninja Assassin|2009|Prepare to enter a secret world of assassins
137|Frost/Nixon|2008|400 million people were waiting for the truth.
121|Speed Racer|2008|Speed has no limits
159|Charlie Wilson's War|2007|A stiff drink. A little mascara. A lot of nerve. Who said they couldn't bring down the Soviet empire.
116|V for Vendetta|2006|Freedom! Forever!
111|The Da Vinci Code|2006|Break The Codes
92|RescueDawn|2006|Based on the extraordinary true story of one man's fight for freedom
161|The Polar Express|2004|This Holiday Season… Believe
154|Something's Gotta Give|2003|
(10 rows)
SELECT * FROM movie m WHERE m.title = 'The Matrix' ORDER BY m.id ASC;
id|title|released|tagline
0|The Matrix|1999|Welcome to the Real World
(1 row)
SELECT * FROM movie m WHERE m.title LIKE '%Matrix%' ORDER BY m.id ASC;
id|title|released|tagline
0|The Matrix|1999|Welcome to the Real World
9|The Matrix Reloaded|2003|Free your mind
10|The Matrix Revolutions|2003|Everything that has a beginning has an end
(3 rows)
SELECT * FROM movie m WHERE m.title ILIKE '%matrix%' ORDER BY m.id ASC;
id|title|released|tagline
0|The Matrix|1999|Welcome to the Real World
9|The Matrix Reloaded|2003|Free your mind
10|The Matrix Revolutions|2003|Everything that has a beginning has an end
(3 rows)
SELECT count(*) FROM movie m;
count
38
(1 row)
SELECT count(*) FROM movie m WHERE m.released > 2000;
count
12
(1 row)
SELECT avg(released) FROM movie m WHERE m.title like '%Matrix%';
avg
2001.6666666666666667
(1 row)
SELECT title, released, tagline FROM movie m WHERE m.title IN ('The Matrix', 'Top Gun') AND m.released > 1980 ORDER BY m.id ASC;
title|released|tagline
The Matrix|1999|Welcome to the Real World
Top Gun|1986|I feel the need, the need for speed.
(2 rows)
SELECT title, released, tagline FROM movie m WHERE NOT m.title IN ('The Matrix', 'Top Gun') AND m.released > 1980 ORDER BY m.id ASC;
title|released|tagline
The Matrix Reloaded|2003|Free your mind
The Matrix Revolutions|2003|Everything that has a beginning has an end
The Devil's Advocate|1997|Evil has its winning ways
A Few Good Men|1992|In the heart of the nation's capital, in a courthouse of the U.S. government, one man will stop at nothing to keep his honor, and one will stop at nothing to find the truth.
Jerry Maguire|2000|The rest of his life begins now.
Stand By Me|1986|For some, it's the last real taste of innocence, and the first real taste of life. But for everyone, it's the time that memories are made of.
As Good as It Gets|1997|A comedy from the heart that goes for the throat.
What Dreams May Come|1998|After life there is more. The end is just the beginning.
Snow Falling on Cedars|1999|First loves last. Forever.
You've Got Mail|1998|At odds in life... in love on-line.
Sleepless in Seattle|1993|What if someone you never met, someone you never saw, someone you never knew was the only someone for you?
Joe Versus the Volcano|1990|A story of love, lava and burning desire.
When Harry Met Sally|1998|At odds in life... in love on-line.
That Thing You Do|1996|In every life there comes a time when that thing you dream becomes that thing you do
The Replacements|2000|Pain heals, Chicks dig scars... Glory lasts forever
RescueDawn|2006|Based on the extraordinary true story of one man's fight for freedom
The Birdcage|1996|Come as you are
Unforgiven|1992|It's a hell of a thing, killing a man
Johnny Mnemonic|1995|The hottest data on earth. In the coolest head in town
Cloud Atlas|2012|Everything is connected
The Da Vinci Code|2006|Break The Codes
V for Vendetta|2006|Freedom! Forever!
Speed Racer|2008|Speed has no limits
Ninja Assassin|2009|Prepare to enter a secret world of assassins
The Green Mile|1999|Walk a mile you'll never forget.
Frost/Nixon|2008|400 million people were waiting for the truth.
Hoffa|1992|He didn't want law. He wanted justice.
Apollo 13|1995|Houston, we have a problem.
Twister|1996|Don't Breathe. Don't Look Back.
Cast Away|2000|At the edge of the world, his journey begins.
Something's Gotta Give|2003|
Bicentennial Man|1999|One robot's 200 year journey to become an ordinary man.
Charlie Wilson's War|2007|A stiff drink. A little mascara. A lot of nerve. Who said they couldn't bring down the Soviet empire.
The Polar Express|2004|This Holiday Season… Believe
A League of Their Own|1992|Once in a lifetime you get a chance to do something different.
(35 rows)
